plugins/alignments/src/CramAdapter/CramAdapter.ts

```diff
@@ -67,62 +68,55 @@ export default class CramAdapter extends BaseFeatureDataAdapter {
       throw new Error('Error getting subadapter')
     }
 
-    const { dataAdapter } = await this.getSubAdapter(
-      readConfObject(this.config, 'sequenceAdapter'),
-    )
-    if (dataAdapter instanceof BaseFeatureDataAdapter) {
-      this.sequenceAdapter = dataAdapter
-    } else {
+    const seqConf = readConfObject(this.config, 'sequenceAdapter')
+    const { dataAdapter: sequenceAdapter } = await this.getSubAdapter(seqConf)
+
+    if (!(sequenceAdapter instanceof BaseFeatureDataAdapter)) {
       throw new Error(
         `CRAM feature adapters cannot use sequence adapters of type '${sequenceAdapterType}'`,
       )
     }
-    return { sequenceAdapter: this.sequenceAdapter }
+
+    return { cram, sequenceAdapter }
   }
 
   async getHeader(opts?: BaseOptions) {
-    await this.configure()
-    return this.cram.cram.getHeaderText(opts)
+    const { cram } = await this.configure()
+    return cram.cram.getHeaderText(opts)
   }
 
   private async seqFetch(seqId: number, start: number, end: number) {
     start -= 1 // convert from 1-based closed to interbase
 
-    const refSeqStore = this.sequenceAdapter
-    if (!refSeqStore) {
-      return undefined
-    }
+    const { sequenceAdapter } = await this.configure()
     const refName = this.refIdToOriginalName(seqId) || this.refIdToName(seqId)
     if (!refName) {
       return undefined
     }
 
-    const features = refSeqStore.getFeatures(
-      {
+    const seqChunks = await sequenceAdapter
+      .getFeatures({
         refName,
         start,
         end,
         assemblyName: '',
-      },
-      {},
-    )
-
-    const seqChunks = await features.pipe(toArray()).toPromise()
+      })
+      .pipe(toArray())
+      .toPromise()
 
-    const trimmed: string[] = []
-    seqChunks
+    const sequence = seqChunks
       .sort((a, b) => a.get('start') - b.get('start'))
-      .forEach((chunk: Feature) => {
+      .map(chunk => {
         const chunkStart = chunk.get('start')
         const chunkEnd = chunk.get('end')
         const trimStart = Math.max(start - chunkStart, 0)
         const trimEnd = Math.min(end - chunkStart, chunkEnd - chunkStart)
         const trimLength = trimEnd - trimStart
         const chunkSeq = chunk.get('seq') || chunk.get('residues')
-        trimmed.push(chunkSeq.substr(trimStart, trimLength))
+        return chunkSeq.substr(trimStart, trimLength)
       })
+      .join('')
```

Comment:
are you sure this doesn't have worse performance than the previous implementation? The map and join seems like it's going to be storing big arrays of some of the intermediate results

Reply:
the current code does a join also, this just rearranged it a bit to avoid a temp variable. but, if i understand you, we could do string append instead. the string append may be a bit faster in some circumstances but i think it might be statistical noise. some benchmarks here show that the join is actually faster than string appends (if strings are not random, then can take advantage of some string pooling and append is faster but i believe our sequences are effectively random)

short random https://jsbench.me/3bkymf6730/1
long random string https://jsbench.me/3bkymf6730/3